test/integration/test_atlas.py

```diff
@@ -0,0 +1,69 @@
+import os
+import random
+
+import lorem
+import pytest
+from pymongo import MongoClient
+
+import superduperdb as s
+from superduperdb import superduper
+from superduperdb.base.config import SelfHosted
+from superduperdb.container.document import Document
+from superduperdb.container.listener import Listener
+from superduperdb.container.model import Model
+from superduperdb.container.vector_index import VectorIndex
+from superduperdb.db.mongodb.query import Collection
+from superduperdb.ext.vector.encoder import vector
+
+ATLAS_VECTOR_URI = os.environ.get('ATLAS_VECTOR_URI')
+
+
+def random_vector_model(x):
+    return [random.random() for _ in range(16)]
+
+
+@pytest.fixture()
+def atlas_search_config():
+    previous = s.CFG.vector_search.type
+    s.CFG.vector_search.type = SelfHosted()
+    yield
+    s.CFG.vector_search.type = previous
+
+
+@pytest.mark.skipif(ATLAS_VECTOR_URI is None, reason='Only atlas deployments relevant.')
+def test_setup_atlas_vector_search(atlas_search_config):
+    model = Model(
+        identifier='test-model', object=random_vector_model, encoder=vector(shape=(16,))
+    )
+    client = MongoClient(ATLAS_VECTOR_URI)
+    db = superduper(client.test_atlas_vector_search)
+    collection = Collection('docs')
+    db.execute(
+        collection.insert_many(
+            [Document({'text': lorem.sentence()}) for _ in range(50)]
+        )
+    )
+    db.add(
+        VectorIndex(
+            'test-vector-index',
+            indexing_listener=Listener(
+                model=model,
+                key='text',
+                select=collection.find(),
+            ),
+        )
+    )
```

Comment:
Comment (perhaps for the future): We should be including `assert` statements in all our tests IMO. I understand that having the code run without error is already useful, but I think we need to 'crossing all the t's' here ie confirming that the test has done the right thing.